frontend/src/scenes/surveys/Survey.tsx

```diff
@@ -28,7 +29,13 @@ export const scene: SceneExport = {
 
 export function SurveyComponent({ id }: { id?: string } = {}): JSX.Element {
     const { isEditingSurvey, surveyMissing } = useValues(surveyLogic)
-    const showSurveyForm = id === 'new' || isEditingSurvey
+    const { editingSurvey } = useActions(surveyLogic)
+
+    useEffect(() => {
+        if (id === 'new') {
+            editingSurvey(true)
+        }
+    }, [id, editingSurvey])
```

Comment:
I reached for `useEffect()` here since that's my classic tool for this type of react state management.  If there's a more posthoggy way to do this, plz let me know.

Reply:
@neilkakkar I might ask for your advice on this.  I know our [coding conventions](https://posthog.com/handbook/engineering/conventions/frontend-coding) say to not use `useEffect` to store state, but in my case here, I can't figure out where to insert the editing process into the survey flow (this is just for new surveys, fwiw).  Let me provide some context.

Right now, the way survey editing works is that whenever we click "New survey" and are prompted with the new survey creation page, we have the `id` prop set to `'new'`, but we haven't entered the survey editing experience yet.  When I tried removing this `useEffect()` and replacing it by just calling the action inside the `Create Blank Survey` button
```
            <PageHeader
                buttons={
                    <LemonButton
                        type="primary"
                        onClick={() => editingSurvey(true)}
                        to={urls.survey('new')}
                        data-attr="new-blank-survey"
                    >
                        Create blank survey
                    </LemonButton>
                }
            />
```

I kept accidentally triggering the `beforeunload` handler, since the editing mode was applied before the new URL loaded.  

Given how surveys are designed now, what I'd need to do is have some sort of way to set editing state _after_ we've landed on the new survey URL.  I'm not sure how to best do that using Kea stuff, so I'd love your input on how to do this the PostHog way.

Right now, what I'm doing works, but it definitely does break our survey editing state management abstraction, and I'm not 100% certain what the appropriate way of resolving this is going forward.

Reply:
Look into `urlToAction` in the surveylogic! I think adding this to the end there should work - it effectively means when you end up on URL X, fire action Y.

And makes sense, on the button would be too soon indeed 👍

Reply:
oh, that's very cool, and sounds like exactly what we're looking for.  Let me give that a shot!

Reply:
Okay, so that _almost_ worked, I made this change to the `urlToAction` hook
```
    urlToAction(({ actions, props }) => ({
        [urls.survey(props.id ?? 'new')]: (_, __, _e__, { method }) => {
            // If the URL was pushed (user clicked on a link), reset the scene's data.
            // This avoids resetting form fields if you click back/forward.
            if (method === 'PUSH') {
                if (props.id) {
                    actions.loadSurvey()
					// new stuff
                    if (props.id === 'new') {
                        actions.editingSurvey(true)
                    }
                } else {
                    actions.resetSurvey()
                }
            }
        },
    })),
```

And it worked for 7/8 of my test cases.  However, it's getting stuck in this loop where if I navigate to an existing survey from the `/surveys` URL, enter editing mode, and then try to hit the "back" button, I'm stuck in an infinite beforeunload loop because it keeps toggling between the 2 URLs (see attached gif).  I'm not sure how to get around it.

At this point, I'm still kinda included to go with `useEffect()` on the survey page because it always works, and then clean up this state management as I keep going.  The idea here is that I fix the broken behavior immediately and then fix the state abstractions once the user pain is mitigated.  What do you think?

![2024-05-23 09 54 57](https://github.com/PostHog/posthog/assets/4853149/a991b516-11b4-4aa9-a738-66226897cfd1)

Reply:
d'oh, annoying, yeah agree sounds good, let's go with the useEffect for now, I'll take another look at this on Monday too - unclear why it would get stuck 😓

Reply:
Oh very lame, this is happening with `useEffect()` too, actually (gif example with the URLs included so you can see what I mean).  The specific use case is when a user goes from surveys page -> clicks on existing survey -> clicks "edit" -> tries specifically to click the "back" button in the browser (which sends them from the `/surveys/{id}` URL to `/surveys`.  

Gonna get back in the lab for this, I don't think this behavior is an acceptable fix.

![2024-05-23 12 11 18](https://github.com/PostHog/posthog/assets/4853149/569a5bba-b242-4882-a64b-d4609376751d)

Reply:
spent some time looking into this, this is too flakey indeed, and I couldn't figure this out. I'm leaning a lot more towards Marius's suggestion now indeed - we can base64 encode the entire payload into the URL, so when you click back, and then forward, the url populates the data again, seems much cleaner 👍